File: internal/analyzer/testdata/python/good_signatures.py
from temporalio import workflow, activity
from dataclasses import dataclass


@dataclass
class WorkflowInput:
    name: str
    age: int
    active: bool


@dataclass
class ActivityInput:
    name: str
    count: int


@workflow.defn
class GoodSignatureWorkflow:
    @workflow.run
    async def run(self, input: WorkflowInput) -> str:
        return f"{input.name}-{input.age}"


@activity.defn
async def good_activity(input: ActivityInput) -> str:
    return f"{input.name}-{input.count}"

File: internal/analyzer/testdata/rust/good_signatures.rs
use temporal_sdk::prelude::*;

#[derive(Serialize, Deserialize)]
pub struct WorkflowInput {
    pub name: String,
    pub age: u32,
}

#[workflow_run]
pub async fn good_workflow_sig(ctx: WfContext, input: WorkflowInput) -> Result<String, anyhow::Error> {
    Ok(format!("{} {}", input.name, input.age))
}

#[activity]
pub async fn good_activity(ctx: ActContext, input: ActivityInput) -> Result<String, anyhow::Error> {
    Ok(input.id)
}
